proto/identity.go

```diff
@@ -48,6 +55,35 @@ type IdentityView struct {
 	ServerEra string `json:"server_era"` // the era of the server that captured this view
 }
 
+// LoadEmoji takes a json key-value object stored in the file at path and
+// unmarshals it into the global validEmoji map[string]string.
+func LoadEmoji(path string) error {
+	raw, err := ioutil.ReadFile(path)
+	if err != nil {
+		return err
+	}
+	if err := json.Unmarshal(raw, &validEmoji); err != nil {
+		return err
+	}
+	return nil
+}
+
+func nickLen(nick string) int {
+	indices := possibleEmoji.FindAllStringIndex(nick, 36)
+	for _, item := range indices {
+		s := nick[item[0]+1 : item[1]-1]
+		_, ok := validEmoji[s]
+		if ok {
+			ret := 1 + nickLen(nick[0:item[0]])
+			if item[1] < len(nick)-1 {
```

Comment:
This undercounts by one if `item[1] == len(nick)-1`, because `item[1]` is the index of the byte following the emoji. Just compare against `len(nick)`. Add a test case for this boundary condition.